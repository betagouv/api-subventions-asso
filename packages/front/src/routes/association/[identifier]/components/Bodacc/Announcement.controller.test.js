import AnnouncementController from "./Announcement.controller";

describe("AnnouncementController", () => {
    const ANNOUNCEMENT = { id: "A202202272158", jugement: '{"complementJugement": "Judgment has text"}' };

    describe("getter judgment", () => {
        it("should return judgment", () => {
            const controller = new AnnouncementController(ANNOUNCEMENT);
            const actual = controller.judgment;
            expect(actual).toMatchSnapshot();
        });
    });

    describe("getter url", () => {
        it("should return url", () => {
            const controller = new AnnouncementController(ANNOUNCEMENT);
            const actual = controller.url;
            expect(actual).toMatchSnapshot();
        });
    });

    describe("getter publication", () => {
        it("should call _splitId()", () => {
            const controller = new AnnouncementController(ANNOUNCEMENT);
            const spySplitId = vi.spyOn(controller, "_splitId");
            controller.publicationFile;
            expect(spySplitId).toHaveBeenCalledTimes(1);
        });

        it("should return url", () => {
            const controller = new AnnouncementController(ANNOUNCEMENT);
            const actual = controller.publicationFile;
            expect(actual).toMatchSnapshot();
        });
    });
});
