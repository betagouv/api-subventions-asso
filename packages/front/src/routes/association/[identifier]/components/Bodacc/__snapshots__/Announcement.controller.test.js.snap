// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AnnouncementController > getter judgment > should return judgment 1`] = `"Judgment has text"`;

exports[`AnnouncementController > getter publication > should return url 1`] = `"https://www.bodacc.fr/telechargements/COMMERCIALES/PDF/A/2022/20220227/1/BODACC_A_PDF_Unitaire_20220227_02158.pdf"`;

exports[`AnnouncementController > getter url > should return url 1`] = `"https://www.bodacc.fr/pages/annonces-commerciales-detail/?q.id=id:A202202272158"`;
